peekingduck/pipeline/nodes/dabble/utils/tracking_files/opencv_tracking.py

```diff
@@ -0,0 +1,157 @@
+            prev_frame_tracked_bbox.append(np.array(value[1]))
+
+        for box in bboxes:
+            # Get matching ious for each bbox in frame to previous bboxes
+            ious = iou(np.array(box), np.array(prev_frame_tracked_bbox))
+            # Check if current bbox passes iou_thresh with any previous
+            # tracked bboxes and get index of highest iou above threshold
+            prev_frame_bbox_highest_iou_index = \
+                ious.argmax() if round(max(ious), 1) >= self.iou_thresh else None
+            matching_dict.update(
+                {tuple(box): prev_frame_bbox_highest_iou_index})
+
+        # Create object tags from highest IOU index and tracking_dict
+        track_id = []
+        for key, value in matching_dict.items():
+            if value is not None:
+                # Get object ID through prev_frame_bbox_highest_iou_index
+                id_num = list(self.tracking_dict)[value]  # type: ignore
+                track_id.append(str(id_num))
+            else:
+                # Create new tracker for bbox that < IOU threshold
+                self._initialize_tracker(key, frame)
+                id_num = list(self.tracking_dict)[-1]
+                track_id.append(str(id_num))
+        # Create result list to replace duplicate track_ids
+        obj_tags = []
+        for id_num in track_id:
+            if id_num not in obj_tags:
+                obj_tags.append(id_num)
+            else:
+                obj_tags.append("")
+
+        return obj_tags
+
+    def _initialize_tracker(self,
+                            bbox: List[float],
+                            frame: np.ndarray) -> Dict[int, List[Any]]:
+        """Start a tracker for each bbox"""
+        tracker = self._create_tracker_by_name(self.tracker_type)
+        tracker.init(frame, tuple(bbox))
+        self.next_object_id += 1
+        self.tracking_dict.update({self.next_object_id: [tracker, bbox]})
+        return self.tracking_dict
+
+    @staticmethod
+    def _create_tracker_by_name(tracker_type: str) -> Any:
+        """Create tracker based on tracker name"""
+        if tracker_type == tracker_types[0]:
+            tracker = cv2.TrackerBoosting_create()
+        elif tracker_type == tracker_types[1]:
+            tracker = cv2.TrackerMIL_create()
+        elif tracker_type == tracker_types[2]:
+            tracker = cv2.TrackerKCF_create()
+        elif tracker_type == tracker_types[3]:
+            tracker = cv2.TrackerTLD_create()
+        elif tracker_type == tracker_types[4]:
+            tracker = cv2.TrackerMedianFlow_create()
+        elif tracker_type == tracker_types[5]:
+            tracker = cv2.TrackerGOTURN_create()
+        elif tracker_type == tracker_types[6]:
+            tracker = cv2.TrackerMOSSE_create()
+        elif tracker_type == tracker_types[7]:
+            tracker = cv2.TrackerCSRT_create()
+        else:
+            tracker = None
+        return tracker
```

Comment:
Is this needed? There seems to be overlapping files? Can you check? Thanks. Please remove redundant files (only leave mosse and iou for now)

Reply:
Removed redundant tracker types